eth/catalyst/api.go

```diff
@@ -756,3 +758,61 @@ func (api *ConsensusAPI) heartbeat() {
 func (api *ConsensusAPI) ExchangeCapabilities([]string) []string {
 	return caps
 }
+
+// GetPayloadBodiesV1 implements engine_getPayloadBodiesByHashV1 which allows for retrieval of a list
+// of block bodies by the engine api.
+func (api *ConsensusAPI) GetPayloadBodiesByHashV1(hashes []common.Hash) []*beacon.ExecutionPayloadBodyV1 {
+	var bodies = make([]*beacon.ExecutionPayloadBodyV1, len(hashes))
+	for i, hash := range hashes {
+		block := api.eth.BlockChain().GetBlockByHash(hash)
+		bodies[i] = getBody(block)
+	}
+	return bodies
+}
+
+// GetPayloadBodiesByRangeV1 implements engine_getPayloadBodiesByRangeV1 which allows for retrieval of a range
+// of block bodies by the engine api.
+func (api *ConsensusAPI) GetPayloadBodiesByRangeV1(start, count uint64) ([]*beacon.ExecutionPayloadBodyV1, error) {
+	if start == 0 || count == 0 || count > 1024 {
+		return nil, beacon.InvalidParams.With(fmt.Errorf("invalid start or count, start: %v count: %v", start, count))
+	}
+	// limit count up until current
+	current := api.eth.BlockChain().CurrentBlock().NumberU64()
+	end := start + count
+	if end > current {
+		end = current
+	}
+	var bodies []*beacon.ExecutionPayloadBodyV1
+	for i := uint64(start); i < end; i++ {
```

Comment:
```suggestion
	for i := start; i < end; i++ {
```
why cast uint64 to uint64?